From: Sergei Volkov <taranarmo@gmail.com>
Subject: [PATCH] llvm: add zlib and zstd build rpath

When building LLVM as part of Julia source build, intermediate build tools
like llvm-min-tblgen dynamically link against zlib/zstd located in build_shlibdir.
Upstream (JuliaLang/julia#63103) added CMAKE_BUILD_RPATH only for Darwin,
so on Linux in the Nix build sandbox these tools fail to load libz.so.1 unless
build_shlibdir is also added to CMAKE_BUILD_RPATH.

--- a/deps/llvm.mk
+++ b/deps/llvm.mk
@@ -108,6 +108,7 @@ LLVM_CMAKE += -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD:STRING="$(LLVM_EXPERIMENTAL_T
 LLVM_CMAKE += -DLLVM_ENABLE_LIBXML2=OFF -DLLVM_HOST_TRIPLE="$(or $(XC_HOST),$(BUILD_MACHINE))"
 LLVM_CMAKE += -DLLVM_ENABLE_ZLIB=FORCE_ON -DZLIB_ROOT="$(build_prefix)"
 LLVM_CMAKE += -DLLVM_ENABLE_ZSTD=FORCE_ON -DZSTD_ROOT="$(build_prefix)"
+LLVM_CMAKE += -DCMAKE_BUILD_RPATH="$(build_shlibdir)"
 ifeq ($(USE_POLLY_ACC),1)
 LLVM_CMAKE += -DPOLLY_ENABLE_GPGPU_CODEGEN=ON
 endif
